File: bankingOnGolang/internal/model/ExternalAccount.go

package model

import (
)

//==============================================================
// ExternalAccount Declaration
//==============================================================
type ExternalAccount struct {
    BaseModel
     Name            string
    Iban            IBAN
    AccountNumber            AccountNumber
    Bic            BIC
    BankName            string
    Country            string
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
     Transactions           []Transaction `gorm:"foreignKey:TransactionsFromExternalAccountId"`

// parent associations as their child

}

File: src/app/models/ExternalAccount.ts
// Define collection and schema for ExternalAccount
export interface ExternalAccount {
  name: string;
  iban: IBAN;
  accountNumber: AccountNumber;
  bic: BIC;
  bankName: string;
  country: string;
  Customer: Schema.Types.ObjectId;
  Transactions: Schema.Types.ObjectId[];
  collection: "externalAccounts";
}
